### PR TITLE
feat: set up DNS for fpx.no

### PR DESCRIPTION
**Pull request by braaar**
### TL;DR
Adds support for using fpx.no subdomains for customers, so that they don't need their own domain/DNS records, etc.

## What's changed
- Allow customers to use *.fpx.no instead of their own custom domain (leave domain field blank in sanity to do this, the ident field is used for subdomain)
- Describe some resources as legacy in the source code, which should be removed once current customers are migrated to fpx.no*
- Add new DNS records for fpx.no subdomains
- Create ingresses for fpx.no, which are the default moving forwards (legacy customers excepted)
- Remove some unnecessary env variables from portal api
- The new domains will look like this:
  - debitor portal: `vasekomaniet.fpx.no`
  - creditor portal: `kred.vasekompaniet.fpx.no`
  - portal api: `api.vasekompaniet.fpx.no`

*We can still support custom domains in the future. However, it would be preferable if we make some tweaks to the code relating to custom domains, which is safe and comfortable to do when there are no customers depending on this feature. That way we can avoid legacy shenanigans (such as customers depending on the bjerk.dev domain, which has caused problems in the past when the bjerk.dev domain expired).

### Why

It is desired to offer a clean looking domain to customers.

### CODE REVIEW

File: resources/kubernetes/portal-api/portal-api.ts [hunk 1]
```diff
@@ -15,8 +14,8 @@ const config = new pulumi.Config('portal-api');
 const authSignSecret = config.requireSecret('auth-sign-secret');
 const cookieSecret = config.requireSecret('cookie-secret');
 
-export const portalApiDevDomain = config.require('dev-domain');
-const cleanPortalApiDomain = portalApiDevDomain.slice(0, -1);
+export const portalApiDomain = config.require('domain');
```
Comment (Xillians) [inline resources/kubernetes/portal-api/portal-api.ts:17]: If we have a root domain (`fpx.no`), do we need to also get this value? 👀 that would be `api.fpx.no` right? We could just apply api as `api.{rootDomain}.no`?

**Review by Xillians — APPROVED** [review]

File: Pulumi.prod.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 config:
-  debitor-portal-app:dev-domain: debitor.flexisoft.bjerk.dev.
+  debitor-portal-app:domain: debitor.fpx.no.
```
Comment (Xillians) [inline Pulumi.prod.yaml:2]: This should be just `fpx.no` right? 🤔

File: resources/kubernetes/portal-app/portal-domains.ts [hunk 3]
```diff
@@ -1,89 +1,135 @@
...
-                        pathType: 'Prefix',
-                        backend: {
-                          service: {
-                            name: portalApp.service.metadata.name,
-                            port: { number: portalApp.port },
-                          },
-                        },
+    const domainKind = customer.domain ? 'customer-domain' : 'root-domain';
+    const debitorPortalDomain = customerDomain;
+    const creditorPortalDomain = `kred.${customerDomain}`;
```
Comment (Xillians) [inline resources/kubernetes/portal-app/portal-domains.ts:20]: is `kred.{customerDomain}` the url we want to use? the pulumi var says `creditor.fpx.no`.

File: resources/kubernetes/portal-app/portal-domains.ts [hunk 4]
```diff
@@ -1,89 +1,135 @@
...
+          },
+          labels: {
+            customer: customer.ident.current,
+            kind: domainKind,
+          },
+        },
+        spec: {
+          rules: [
+            {
+              host: `kred.${customerDomain}`,
```
Comment (Xillians) [inline resources/kubernetes/portal-app/portal-domains.ts:79]: is `kred.{customerDomain}` the url we want to use? the pulumi var says `creditor.fpx.no`.

File: Pulumi.prod.yaml (hunk 2 above)
Comment (braaar) [reply Pulumi.prod.yaml:2]: I think we should reserve this domain for a website or whatever else. This domain won't be used to access an app in itself
